compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -18925,21 +18925,28 @@ private STNode getBindingPattern(STNode ambiguousNode) {
                 STNode varName = ((STSimpleNameReferenceNode) ambiguousNode).name;
                 return createCaptureOrWildcardBP(varName);
             case QUALIFIED_NAME_REFERENCE:
+                if (isListBP) {
+                    STNode identifier = SyntaxErrors.createMissingTokenWithDiagnostics(SyntaxKind.IDENTIFIER_TOKEN,
+                            ParserRuleContext.BINDING_PATTERN_STARTING_IDENTIFIER);
+                    identifier = SyntaxErrors.cloneWithLeadingInvalidNodeMinutiae(identifier, ambiguousNode);
+                    return createCaptureOrWildcardBP(identifier);
```

Comment:
Instead of doing the validation here, can't we execute the same validation done at the default?
You can take the default case logic out of the switch and `break` here?
```suggestion
                    break;
```

Reply:
Updated 8361199b43b0120ab8d1bf3af49d9c97d51289be